File: HorseRacing.Contracts/Models/Game/Dtos/JoinGameWithBetRequestDto.cs
using HorseRacing.Contracts.Base.Dto;

namespace HorseRacing.Contracts.Models.Game.Dtos
{
    public class JoinGameWithBetRequestDto : BaseDto
    {
        /// <summary>
        /// Код пользователя
        /// </summary>
        public Guid UserId { get; set; }
        /// <summary>
        /// Код игры
        /// </summary>
        public Guid GameId { get; set; }
        /// <summary>
        /// Ставка
        /// </summary>
        public int BetAmount { get; set; }
        /// <summary>
        /// Масть
        /// </summary>
        public int BetSuit { get; set; }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/game/requests/join-game-with-bet/join-game-with-bet-request-dto.ts
import type { BaseDto } from "~/interfaces/api/contracts/base/dto/base-dto";

export interface JoinGameWithBetRequestDto extends BaseDto {
    /** Идентификатор игры */
    GameId: string;
    /** Идентификатор пользователя */
    UserId: string;
    /** Ставка пользователя */
    BetAmount: number;
    /** Масть лошади, которую выбрал пользователь */
    BetSuit: number;
}
